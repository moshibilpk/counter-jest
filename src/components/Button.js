
import { TouchableHighlight, Text, StyleSheet } from 'react-native'
import React from 'react'

type PropsType = {|
    onPress?: () => Promise <void> | void,
    title: string | Element <typeof Text>,
  |};

const Button = ({ buttonName, onPress }:PropsType): Element<typeof TouchableHighlight>  => {
    return (
        <TouchableHighlight style={styles.counterButton} onPress={onPress}>
            <Text style={styles.labelStyle}>{buttonName}</Text>
        </TouchableHighlight>
    )
}

export default Button

const styles = StyleSheet.create({
    counterButton: {
        backgroundColor: '#000',
        padding: 10,
        borderRadius: 5
    },
    labelStyle: {
        color: '#fff'
    }
})

