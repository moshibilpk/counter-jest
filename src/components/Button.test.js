// @flow

import React from 'react';
import {Text, TouchableHighlight} from 'react-native';
import {findAllWithType, findWithType} from 'react-shallow-testutils';
import ShallowRenderer from 'react-test-renderer/shallow';
import Button from './Button';

const renderer = new ShallowRenderer();

let buttonName: string, onPress: () => void;

describe('button for counter', () => {
  beforeEach(() => {
    buttonName = 'buttonName';
    onPress = jest.fn();
  });

  test('renders correctly', () => {
    renderer.render(<Button onPress={onPress} buttonName={buttonName} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  test('renders button with correct name', () => {
    renderer.render(<Button onPress={onPress} buttonName={buttonName} />);
    const result = renderer.getRenderOutput();
    const text = findWithType(result, Text);
    expect(text.props.children).toBe(buttonName);
  });

  it('renders button with correct action', () => {
    renderer.render(<Button onPress={onPress} buttonName={buttonName} />);
    const result = renderer.getRenderOutput();
    const button = findWithType(result, TouchableHighlight);
    button.props.onPress();
    expect(typeof button.props.onPress).toBe('function');
  });
});
