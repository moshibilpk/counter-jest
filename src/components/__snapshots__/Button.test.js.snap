// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button for counter renders correctly 1`] = `
<TouchableHighlight
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#000",
      "borderRadius": 5,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    buttonName
  </Text>
</TouchableHighlight>
`;
